src/vm/functions/mod.rs

```diff
@@ -25,16 +25,40 @@ fn native_eq(args: &[Value]) -> Result<Value> {
     }
 }
 
+fn native_hash160(args: &[Value]) -> Result<Value> {
+    use util::hash::Hash160;
+
+    if !(args.len() == 1) {
+        return Err(Error::new(ErrType::InvalidArguments("Wrong number of arguments to hash160 (expects 1)".to_string())))
+    }
+    let input = &args[0];
+    let bytes = match input {
+        Value::Int(value) => Ok(value.to_le_bytes().to_vec()),
+        Value::Buffer(value) => Ok(value.data.clone()),
+        _ => Err(Error::new(ErrType::NotImplemented))
+    }?;
+    let hash160 = Hash160::from_data(&bytes);
+    Value::buff_from(hash160.as_bytes().to_vec())
+}
+
 fn native_begin(args: &[Value]) -> Result<Value> {
     match args.last() {
         Some(v) => Ok(v.clone()),
         None => Ok(Value::Void)
     }
 }
 
+fn native_print(args: &[Value]) -> Result<Value> {
```

Comment:
A smart contract can `print`?  Where would this data end up -- the node's log file?

Reply:
Would recommend making this function a no-op unless develop or debug mode was enabled on the node.  For example, an attacker can do nasty things like insert VT100 control code characters into the log file this way, causing bad things to happen in the victim's terminal.  I'd recommend further extending this function so that when it does print something, we sanitize it first (i.e. convert non-ASCII characters to some ASCII representation).

Reply:
Oh, yeah, definitely this should be a no-op if the node isn't in develop mode.

Reply:
I tried playing around with this some more.  Do you think we will want helpers like `print-str!` that will treat a buffer as a string of ASCII characters, escape non-printable ones, and write those out?  Same idea goes for `print-int!` or `print-hex!`.